runtime/src/main/java/org/corfudb/runtime/CheckpointLivenessUpdater.java

```diff
@@ -29,6 +32,8 @@ public class CheckpointLivenessUpdater implements LivenessUpdater {
 
     private final Table<TableName, ActiveCPStreamMsg, Message> activeCheckpointsTable;
 
+    private final Lock currentTableLock = new ReentrantLock();
```

Comment:
I think this complicates the logic and blocks the checkpointer. To solve the concurrency issue, we can just save a copy of `currentTable` at the beginning of `updateHeartbeat`. Besides the happy pass where `currentTable` is not changed during `updateHeartbeat`, it either hits transaction conflict (if the entry has been just deleted by `unlockTableAfterCheckpoint`), or gets an empty record (if the `currentTable` has been changed right after the copy)